File: Selecao I/impar.cpp
/*@impar: Ímpar ou par - OBI 2016 - F1P1*/

#include <iostream>
#include <locale>
using namespace std;

int main(){
    setlocale(LC_ALL, "Portuguese");
    int p, d1, d2;
    
    cin >> p;
    cin >> d1;
    cin >> d2;

    int soma = d1 + d2;

    if ((soma % 2 == 0 && p == 0) || (soma % 2 != 0 && p == 1)) {
      cout << 0 << endl; 
    } else {
      cout << 1 << endl; 
    }
    return 0;
}

File: src/Selecao_II/Impar.java
/**
 * Questão: @impar: Ímpar ou par - OBI 2016 - F1P1
 *
 * @author UxieGu1
 */
package Selecao_II;
import java.util.Scanner;

public class Impar {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int p = scan.nextInt();
        int d1 = scan.nextInt();
        int d2 = scan.nextInt();

        int soma = d1 + d2;

        if ((soma % 2 == 0 && p == 0) || (soma % 2 != 0 && p == 1)) {
            System.out.println(0);
        } else {
            System.out.println(1);
        }

        scan.close();
    }
}
